evalml/tests/automl_tests/test_automl.py

```diff
@@ -2579,6 +2604,7 @@ def test_automl_one_allowed_pipeline_ensembling_disabled(
         ensembling=True,
     )
     automl.search()
+    automl._SLEEP_TIME = 0.0001
```

Comment:
@freddyaboulton do you think there's a way to define a fixture for automl tests which handles stuff like `_SLEEP_TIME`, mocking pipeline methods and stuff like `Optimizer.tell`, etc?

Reply:
@dsherry Good point! Per our discussion at stand-up today, I'll merge this as-is but then look into a way we can package our best practices into a fixture!

Reply:
Sounds great! Sorry for any confusion, this comment is definitely not blocking this PR, and we can discuss after this PR is merged.